### PR TITLE
Document the current index variable

### PR DESCRIPTION
**Pull request by WimJongeneel**

### CODE REVIEW

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ A loop is used to execute repeated actions and is visualized as a frame. For eac
 
 The loop can contain all types of elements used in microflows, except for [start events](/refguide/start-event/) and [end events](/refguide/end-event/). Only a loop can contain [break events](/refguide/break-event/) and [continue events](/refguide/continue-event/).
 
+## The current index variable
+
+Within a loop body, the variable `$currentIndex` contains the current index of the loop. This index starts at zero and increments for each iteration. When a loop is nested inside another loop, the `$currentIndex` variable holds the index of its direct parent. The `$currentIndex` variable is also available inside the expression of a while loop. This feature was introduced in Mendix 10.17.
```
Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md]: >When a loop is nested inside another loop, the `$currentIndex` variable holds the index of its direct parent.

This part is confusing. I read this as, inner loops don't have their own `$currentIndex`. Meaning, it only works for the most outer loop. But what I think you mean is that the `$currentIndex` is always specific to the iteration of the loop action it is used in?

Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md]:
```suggestion
Within a loop, the variable `$currentIndex` contains the current index of the loop. This index starts at zero and increments for each iteration. When a loop is nested inside another loop, the `$currentIndex` variable holds the index of its direct parent. The `$currentIndex` variable is also available inside the expression of a while loop. This feature was introduced in Mendix 10.17.
```

Comment (Robinfr) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md]: >current index of the loop.

This is a bit weird in the context of the while loop. In fact, I'm not sure if this is clear enough for developers not familiar with these terms. Maybe we can specify it as "how many iterations the loop has gone through so far"?

**Review by Robinfr — CHANGES_REQUESTED** [review]

**Review by Robinfr — APPROVED** [review]

**Comment by Robinfr** [comment]
Can be reviewed by tech writers. 👍

File: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md [hunk 2]
```diff
@@ -10,6 +10,14 @@ A loop is used to execute repeated actions and is visualized as a frame. For eac
 
 The loop can contain all types of elements used in microflows, except for [start events](/refguide/start-event/) and [end events](/refguide/end-event/). Only a loop can contain [break events](/refguide/break-event/) and [continue events](/refguide/continue-event/).
 
+## Loop Variable
+
+{{% alert color="info" %}}
+The `$currentIndex` variable was introduced in Mendix 10.17.0.
+{{% /alert %}}
+
+Within a loop, the `$currentIndex` variable contains the current number of loop iterations. This index starts at zero and increments with each iteration. For nested loops, `$currentIndex` holds the index of the innermost loop it is part of. `$currentIndex` is also available when entering an expression for a [While](#while) loop.
```
Comment (Yiyun333) [inline content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md]: @WimJongeneel @Robinfr Thanks for the PR!
I've reviewed the language. Please have a look at my [changes](https://github.com/mendix/docs/pull/8680/commits/91575a496feeff73af386775d94969316c317d3d) and let me know if they are okay.

Question: What does "it" refer to in the following sentence?
> For nested loops, `$currentIndex` holds the index of the innermost loop **it** is part of.

Can you please clarify a bit? Thanks!

Comment (Yiyun333) [reply content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/loop.md]: Answer provided on Slack.

**Review by Yiyun333 — APPROVED** [review]

**Pull request merged**